### ISSUE TITLE
False negative on reportUnboundVariable

### ISSUE THREAD

**Issue by jfmend** [microsoft/pyright#7879]
## Environment data
Tested on:
- pylance v2024.5.1 
- python 3.10.12

settings.json:
```json
{
    "python.analysis.typeCheckingMode": "basic",
    "python.analysis.diagnosticSeverityOverrides": {
        "reportUnboundVariable": true,
    },
}
```

## Code Snippet

```python
import random
if random.random() < 0.5:
    a = 1
print(a)
```

In the last line, the type hint for `a` is `(variable) a: Unbound | Literal[1]`.

## Expected behavior

An error message at the first use of the possibly unbound variable `a`.

## Actual behavior

No error message.

**Comment by erictraut** [comment]
There are separate `reportUnboundVariable` and `reportPossiblyUnboundVariable` checks. You've enabled the former, but this error is reported under the latter. It's "possibly unbound" in this case because some of the time it will not be unbound. Pyright users requested that I separate these into two different diagnostic checks because the "possibly unbound" is sometimes a false positive, but the "definitely unbound" is rarely a false positive.

**Comment by jfmend** [comment]
Thank you for the explanation.
I thought that I was using the right setting because https://code.visualstudio.com/docs/python/settings-reference does not mention `reportPossiblyUnboundVariable` and on `reportUnboundVariable` says:

> reportUnboundVariable - Diagnostics for unbound and possibly unbound variables.

So I guess de documentation has not been updated yet.

**Comment by erictraut** [comment]
Yes, that documentation appears to be out of date. It is maintained by the pylance team. cc @rchiodo, @debonte 

The [pyright documentation](https://microsoft.github.io/pyright/#/configuration?id=main-configuration-options) is up to date.

**Comment by jfmend** [comment]
Oh great! I didn't know about this pyright  documentation.
Sorry to bother you and thank you very much once again.
